client/state-db/src/lib.rs

```diff
@@ -97,12 +97,19 @@ impl<
 {
 }
 
-/// Backend database trait. Read-only.
+/// Backend database trait.
 pub trait MetaDb {
 	type Error: fmt::Debug;
 
 	/// Get meta value, such as the journal.
 	fn get_meta(&self, key: &[u8]) -> Result<Option<DBValue>, Self::Error>;
+
+	/// Set meta value
+	fn set_meta<V: AsRef<[u8]>>(
```

Comment:
Should rather be called `commit_meta` to emphasise that the transaction is committed immediately.

Reply:
Can we have those as `read_meta` and `write_meta` then?
The word `commit` bears a very strong association with a typical transaction (as in write-write-write — commit or rollback) in my opinion. I do not insist though.

Reply:
> The word commit bears a very strong association with a typical transaction (as in write-write-write — commit or rollback) in my opinion. I do not insist though.

That's the point. Normally StateDb does not write anything to the DB on its own. Functions like "insert_block" do not really insert anything into the database. They return a changeset that is later added to a transaction that is committed to the database. `commit_meta` is an exception to this design.

Reply:
Okay, `set` — isn't an aptly chosen word here. Wouldn't `write_meta`  be clear enough?

Reply:
This seems to be irrelevant In the light of that: https://github.com/paritytech/substrate/pull/10995#discussion_r865700204.